tuxemon/technique.py

```diff
@@ -297,25 +297,38 @@ def changestats(self, user: Monster, target: Monster) -> EffectResult:
         '''
         statsmaster = [self.statspeed, self.stathp, self.statarmour, self.statmelee, self.statranged, self.statdodge]
         statslugs = ['speed', 'hp', 'armour', 'melee', 'ranged', 'dodge']
+        newstatvalue = 0
         for stat, slug in zip(statsmaster, statslugs):
             if not stat:
                 continue
-            value = stat['value']
-            dividing = stat['dividing']
+            minvalue = stat['minvalue']
+            maxvalue = stat['maxvalue']
+            iterator = stat['iterator']
             override = stat['overridetofull']
             basestatvalue = getattr(target, slug)
-            if value > 0 and override == False:
-                if dividing == False:
-                    newstatvalue = basestatvalue + value
-                else:
+            if maxvalue:
+                value = random.randint(minvalue,maxvalue)
```

Comment:
Using random directly makes everything non deterministic. Maybe we should discuss having something similar to numpy `RandomState` objects, in order to be able to reproduce combats, for networking, debugging or rewatching purposes.

Reply:
going to into a later PR